packages/grafana-ui/src/utils/typeahead.ts

```diff
@@ -8,8 +8,8 @@ export const flattenGroupItems = (groupedItems: CompletionItemGroup[]): Completi
       label,
       kind: CompletionItemKind.GroupTitle,
     };
-    all.push(titleItem, ...items);
-    return all;
+
+    return all.concat(titleItem, items);
```

Comment:
nit: If we are starting to optimize things for performance and we are talking about really big lists. I would recommend changing from concat to the `apply` pattern.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply#Examples

Reply:
The reason this was done wasn't primarily performance but the fact that the maximum call stack size is exceeded if `items` is above a certain length. Still - thanks, if this PR moves ahead, I'll evaluate whether `apply` would also work and provides better performance.